File: lib-util/src/main/java/code/util/EncryptedPreference.java
package code.util;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.security.crypto.EncryptedSharedPreferences;
import androidx.security.crypto.MasterKeys;

public class EncryptedPreference {

    private static final String TAG = EncryptedPreference.class.getSimpleName();

    protected static SharedPreferences getSharedPreferences(String fileName) {
        SharedPreferences sharedPreferences = null;
        try {
            String masterKeyAlias = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC);
            sharedPreferences = EncryptedSharedPreferences.create(
                    fileName,
                    masterKeyAlias,
                    AppGlobal.getApplication(),
                    EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
                    EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
            );
        } catch (Exception e) {
            LogUtil.e(TAG, e, "Fail to create encrypted shared preference");
            sharedPreferences = AppGlobal.getApplication().getSharedPreferences(fileName, Context.MODE_PRIVATE);
        }
        return sharedPreferences;
    }

}

File: lib-util/src/main/java/code/util/EncryptedPreference.kt
package code.util

import android.content.Context
import android.content.SharedPreferences
import androidx.security.crypto.EncryptedSharedPreferences
import androidx.security.crypto.MasterKeys
import code.util.AppGlobal.getApplication

open class EncryptedPreference {
    companion object {
        private val TAG = EncryptedPreference::class.java.simpleName

        @JvmStatic
        protected fun getSharedPreferences(fileName: String): SharedPreferences {
            return try {
                val masterKeyAlias = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC)
                EncryptedSharedPreferences.create(
                    fileName,
                    masterKeyAlias,
                    getApplication(),
                    EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
                    EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
                )
            } catch (e: Exception) {
                e.printStackTrace()
                LogUtil.e(TAG, e, "Fail to create encrypted shared preference")
                getApplication().getSharedPreferences(fileName, Context.MODE_PRIVATE)
            }
        }
    }
}